src/Q1.c

```diff
@@ -47,53 +45,58 @@ void * thread_func(void *arg){
     sprintf(temp,"%ld",tid);
     strcat(privateFifo,temp);
 
+    // open private fifo
     int fd_priv;
-    do{
-        fd_priv = open(privateFifo, O_WRONLY);
-    }while(fd_priv==-1);
+    do{fd_priv = open(privateFifo, O_WRONLY);}while(fd_priv==-1);
     if (fd_priv < 0) {
-        //perror("[Server]Error opening private FIFO"); 
-        printRegister(elapsedTime(), threadi, pid, tid, dur, place, GAVUP);
+        printRegister(elapsedTime(), threadi, pid, pthread_self(), dur, place, GAVUP);
```

Comment:
nunca vai entrar neste if, se for -1 imprimimos logo o GAVUP (e depois podemos tentar outra vez idk)

Reply:
Estás certo, mas nós temos um while(fd==-1), ou seja vai ficar nesse ciclo até conseguir abrir, nunca vai entrar nesse if

Reply:
Acho que o melhor seria tentar abrir o private fifo durante um determinado tempo e se após esse tempo retornar -1 então imprimimos o GAVUP e terminamos a thread

Reply:
Concordo, vou por um @ ao stor no slack para saber com quanto tempo é que devemos pôr a esperar